File: Python/Problem_099.py
import math

def euler_99(path="base_exp.txt"):
    best_line = -1
    best_score = float("-inf")

    with open(path, "r") as f:
        for i, line in enumerate(f, start=1):
            b_str, e_str = line.strip().split(",")
            b = int(b_str)
            e = int(e_str)
            score = e * math.log(b)  # ln(b)
            if score > best_score:
                best_score = score
                best_line = i

    return best_line

if __name__ == "__main__":
    print(euler_99("base_exp.txt"))  # 709 for Euler dataset

File: Java/Problem_099.java
import java.io.*;

public class Euler99 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new FileReader("base_exp.txt"));

        int bestLine = -1;
        double bestScore = Double.NEGATIVE_INFINITY;

        String line;
        int i = 0;
        while ((line = br.readLine()) != null) {
            i++;
            int comma = line.indexOf(',');
            long b = Long.parseLong(line.substring(0, comma));
            long e = Long.parseLong(line.substring(comma + 1));

            double score = e * Math.log(b); // ln(b)
            if (score > bestScore) {
                bestScore = score;
                bestLine = i;
            }
        }
        br.close();

        System.out.println(bestLine); // 709 for Euler dataset
    }
}
